tests/test_utils.py

```diff
@@ -374,6 +374,27 @@ def _second_callback(result):
     assert second_results == [3]
 
 
+@pytest.mark.asyncio
+async def test_concurrent_runner_try_put():
```

Comment:
I see another good test here that verifies, that `try_put` will succeed after the task actually finishes.

Reply:
Yes, as we discussed yesterday, I will add more tests

Reply:
I updated the test here: https://github.com/elastic/connectors/pull/1479/commits/8d033e040007d5ada52a6e9661fca170175e3555

I think the scenario you mentioned yesterday is not practical:
When the pool is full and another task is waiting by calling `put`, I don't think another task can jump the queue by calling `try_put`, because `try_put` will check `self.locked()` first, which will [take those waiting tasks into consideration](https://github.com/python/cpython/blob/9e56eedd018e1a4681789e634016cbb7699dcb8a/Lib/asyncio/locks.py#L358-L361).